htdocs/core/modules/DolibarrModules.class.php

```diff
@@ -2244,26 +2244,17 @@ public function delete_module_parts()
 		global $conf;
 
 		$err = 0;
-		$entity = $conf->entity;
 
-		if (is_array($this->module_parts) && !empty($this->module_parts)) {
-			foreach ($this->module_parts as $key => $value) {
-				// If entity is defined
-				if (is_array($value) && isset($value['entity'])) {
-					$entity = $value['entity'];
-				}
-
-				$sql = "DELETE FROM ".MAIN_DB_PREFIX."const";
-				$sql .= " WHERE ".$this->db->decrypt('name')." LIKE '".$this->db->escape($this->const_name)."_".strtoupper($key)."'";
-				$sql .= " AND entity = ".((int) $entity);
+		$sql = "DELETE FROM ".MAIN_DB_PREFIX."const";
+		$sql .= " WHERE ".$this->db->decrypt('name')." LIKE '".$this->db->escape($this->const_name)."_%'"; // remove all MAIN_MODULE_MYMODULE_*
+		$sql .= " AND entity IN (0, ".((int) $conf->entity); // remove global and current entity
```

Comment:
This means when you disable a module in entity 3, you delete conf that may be required by all entities (if they are on 0).
May be that we can add entity 0 only if conf->entity = 1 
So when acting on entity 3, stay on entity 3 only ?

Reply:
@eldy 
I corrected, on the other hand if we use "entity" with a value of 0, in general there are modules that we cannot deactivate in the child entities!